django_extensions/management/commands/dumpscript.py

```diff
@@ -70,8 +71,12 @@ def orm_item_locator(orm_obj):
 
     for key in clean_dict:
         v = clean_dict[key]
-        if v is not None and not isinstance(v, (six.string_types, six.integer_types, float, datetime.datetime)):
-            clean_dict[key] = six.u("%s" % v)
+        if v is not None:
+            if isinstance(v, datetime.datetime):
+                v = v.replace(tzinfo=get_current_timezone())  # get tzinfo from settings.py
+                clean_dict[key] = dateutil.parser.parse(v.isoformat())
```

Comment:
I still wonder why you need to reparse it... Why not directly `clean_dict[key] = v.isoformat()`?

Reply:
It's been a while since I looked at this but indeed there is no reason to reparse it here.

The idea was to do exactly the same thing as elsewhere in the `dumpscript` code. That is have the resulting script do the parsing back from a isoformat to a python datetime object.

Now it's using `clean_dict[key] = dateutil.parser.parse(v.isoformat())` which results in a string like this in the dumpscript for ` datetime.datetime.now().replace(tzinfo=pytz.utc)`:

```python
importer.locate_object(orgklass, "orgid", klass, "pk_name", 123,  {'test': datetime.datetime(2019, 4, 28, 22, 24, 1, 827687, tzinfo=tzutc())})
```

Which I think will raise an `NameError` since `tzutc` is never imported only `tzoffset` is.

What we want is something that looks like this:
```
importer.locate_object(orgklass, "orgid", klass, "pk_name", 123,  {'test': dateutil.parser.parse("2019-04-28T22:26:57.248346+00:00")}
```

Then `locate_object` gets a datetime object instead of just the string with isoformat time in it.

That said... I'm not sure if (and if so how much) this is actually used in the wild.. and the isoformatted string might be 'good enough' if nobody is exactly using the `obj_content` for `locate_object`.

Reply:
I misunderstood. :sweat_smile:

Reply:
I misunderstood. 😅 @trbs